app/javascript/schools/courses/components/curriculum_editor/CurriculumEditor__ContentBlockCreator.re

```diff
@@ -198,34 +213,75 @@ let button = (target, aboveContentBlock, send, addContentBlockCB, blockType) =>
   </label>;
 };
 
+let handleVimeoVideoUpload = (file, vimeo_video) => {
+  Js.log(vimeo_video)
+  let uploadUrl = vimeo_video##uploadUrl
+
+  tusUpload(file, ~uploadUrl)
+  Js.Promise.resolve()
+};
+
 let uploadFile =
-    (target, send, addContentBlockCB, isAboveContentBlock, formData) =>
-  Api.sendFormData(
-    "/school/targets/" ++ (target |> Target.id) ++ "/content_block",
-    formData,
-    json => {
-      Notification.success("Done!", "File uploaded successfully.");
-      let contentBlock = json |> ContentBlock.decode;
-      addContentBlockCB(contentBlock);
-      send(FinishSaving(isAboveContentBlock));
-    },
-    () => send(FailToUpload),
-  );
+    (target, send, addContentBlockCB, aboveContentBlock, blockType, file, formData) => {
+    let isAboveContentBlock = aboveContentBlock != None
+    switch (blockType) {
+    |`File
+    |`Image =>
+      Api.sendFormData(
+        "/school/targets/" ++ (target |> Target.id) ++ "/content_block",
+        formData,
+        json => {
+          Notification.success("Done!", "File uploaded successfully.");
+          let contentBlock = json |> ContentBlock.decode;
+          addContentBlockCB(contentBlock);
+          send(FinishSaving(isAboveContentBlock));
+        },
+        () => send(FailToUpload),
+      );
+    |`VideoEmbed =>
+      let size = file##size
+      CreateVimeoVideo.make(~size, ())
+      |> GraphqlQuery.sendQuery
+      |> Js.Promise.then_(result => {
+	   let vimeo_video = result##createVimeoVideo##vimeoVideo
+           handleVimeoVideoUpload(file, vimeo_video)
+	 })
+      |> Js.Promise.catch(_ => {
+	   send(FailedToCreate);
+	   Js.Promise.resolve();
+      })
+      |> ignore;
+
+      Api.sendFormData(
+        "/school/targets/" ++ (target |> Target.id) ++ "/content_block",
+        formData,
+        json => {
+          Notification.success("Done!", "File uploaded successfully.");
+          let contentBlock = json |> ContentBlock.decode;
+          addContentBlockCB(contentBlock);
+          send(FinishSaving(isAboveContentBlock));
+        },
+        () => send(FailToUpload),
+      );
```

Comment:
This appears to be a duplicate block of code. Wasn't this already handled by the `` `File | `Image =>`` case?

Reply:
Oh, I was going to remove that all together.